packages/ui-components/src/components/calendar-range-dates-selector/calendar-range-dates-selector.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core';
+import { fromDateToISO, fromISOToMoment, getDatesBetweenRange, isDateBefore, isDateSame } from '../../utils/date.helper';
+import { IClickDateEvent } from '../calendar/calendar.types';
+import { ICalendarRangeDatesSelector, ICalendarRangeDatesSelectorEvents, ISelectRangeDates } from './calendar-range-dates-selector.types';
+
+@Component({
+	tag: 'kv-calendar-range-dates-selector',
+	styleUrls: {
+		night: 'calendar-range-dates-selector.night.scss',
+		light: 'calendar-range-dates-selector.light.scss'
+	},
+	shadow: true
+})
+export class KvCalendarRangeDatesSelector implements ICalendarRangeDatesSelector, ICalendarRangeDatesSelectorEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: false }) selectedRangeDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) initialDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) disabledDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) minDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) maxDate?: string;
+
+	/** @inheritdoc */
+	@Event() selectRangeDates: EventEmitter<ISelectRangeDates>;
+
+	private onClickDate = ({ detail: { event, payload: date } }: CustomEvent<IClickDateEvent>): void => {
+		const clickedDateMoment = fromISOToMoment(date);
+
+		const [selectedStartDate, selectedEndDate] = this.getSelectedRangeDates();
+
+		// check if start date is not selected
+		if (selectedStartDate === undefined) {
+			this.selectRangeDates.emit({ event, payload: [clickedDateMoment.toISOString()] });
+			return;
+		}
+
+		// check if clicked date is the same as the selected date
+		if (isDateSame(clickedDateMoment, selectedStartDate)) {
+			// check if end date is not selected
+			if (selectedEndDate === undefined) {
+				this.selectRangeDates.emit({ event, payload: [clickedDateMoment.toISOString(), clickedDateMoment.toISOString()] });
```

Comment:
Since we don't define hours/minutes I don't know if is cool to set the payload with it (by ISOString). Can we discuss more about this? Possibly we can follow this approach now, and ignore in the other side (if is really needed).

Reply:
The thing is that this date will be a datetime in the future. But for now, I think it makes sense to ignore the "time". Will change on the upcoming commit.